addons/mail/static/src/models/composer/composer.js

```diff
@@ -665,9 +665,11 @@ function factory(dependencies) {
             this.update({
                 suggestedChannels: [[
                     'insert-and-replace',
-                    mentions.map(data =>
-                        this.env.models['mail.thread'].convertData(data))
-                    ]],
+                    mentions.map(data => {
+                        const threadData = this.env.models['mail.thread'].convertData(data);
+                        return Object.assign(threadData, { model: 'mail.channel' });
+                    })
```

Comment:
This should be done in reverse, `model` should be a default value, not overriding `convertData` result.
See first line of `_handleNotificationPartnerChannel` as example.